src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpEnvironmentProxy.cs

```
@@ -168,18 +178,77 @@ private static Uri GetUriFromString(string value)
+                }
+                catch { };
+
+                value = value.Substring(separatorIndex + 1);
+                separatorIndex = auth.IndexOf(':');
+                if (separatorIndex == -1)
+                {
+                    user = auth;
+                }
+                else
+                {
+                    user = auth.Substring(0, separatorIndex);
+                    password = auth.Substring(separatorIndex + 1);
+                }
+            }
 
-            if (!value.Contains("://"))
+            int ipV6AddressEnd = value.IndexOf(']');
+            separatorIndex = value.LastIndexOf(':');
+            // No ':' or it is part of IPv6 address.
+            if (separatorIndex == -1 || (ipV6AddressEnd != -1 && separatorIndex < ipV6AddressEnd))
             {
-                value = "http://" + value;
+                host = value;
+            }
+            else
+            {
+                try
+                {
+                    host = value.Substring(0, separatorIndex);
+                    port = Convert.ToInt32(value.Substring(separatorIndex + 1));
+                }
+                catch
+                {
+                    return null;
+                }
             }
 
-            if (Uri.TryCreate(value, UriKind.Absolute, out Uri uri) &&
-                    (uri.Scheme == Uri.UriSchemeHttp || uri.Scheme == Uri.UriSchemeHttps))
+            try
             {
-                // We only support http and https for now.
-                return uri;
+                UriBuilder ub = new UriBuilder("http", host, port);
+                if (user != null)
+                {
+                    ub.UserName = Uri.EscapeDataString(user);
+                }
+                if (password != null)
+                {
+                    ub.Password = Uri.EscapeDataString(password);
+                }
+                return ub.Uri;
             }
+            catch { };
```

I'm not sure if catching errors here is a good idea: if the environment variables are present but can cause exceptions here it will be a silent failure, even adding logging doesn't seem enough. It seems that here we do want to let the exceptions go to the caller and so on.

---

the handler is already explicitly ignoring proxy failures and that matches existing behavior.
I tested this with garbage or unknown scheme and curl handler will just ignore it.

---

Ok then.